src/Nova/Signature.php

```diff
@@ -0,0 +1,122 @@
+use Tipoff\Support\Nova\BaseResource;
+
+class Signature extends BaseResource
+{
+    public static $model = \Tipoff\Waivers\Models\Signature::class;
+
+    public static function label()
+    {
+        return 'Waiver Signatures';
+    }
+
+    public static function singularLabel()
+    {
+        return 'Signature';
+    }
+
+    public static $title = 'id';
+
+    public static $search = [
+        'id',
+    ];
+
+    public static function indexQuery(NovaRequest $request, $query)
+    {
+        if ($request->user()->hasRole([
+            'Admin',
+            'Owner',
+            'Accountant',
+            'Executive',
+            'Reservation Manager',
+            'Reservationist',
+        ])) {
+            return $query;
+        }
+
+        return $query->whereHas('room', function ($roomlocation) use ($request) {
+            return $roomlocation->whereIn('location_id', $request->user()->locations->pluck('id'));
+        });
+    }
+
+    public static $group = 'Reporting';
+
+    public function fieldsForIndex(NovaRequest $request)
+    {
+        return [
+            ID::make()->sortable(),
+            Text::make('Name', function () {
+                return $this->name.' '.$this->name_last;
+            }),
+            BelongsTo::make('Participant', 'participant', nova('participant'))->sortable(),
+            BelongsTo::make('Room', 'room', nova('room'))->sortable(),
+            DateTime::make('Signed', 'created_at')->sortable(),
+        ];
+    }
+
+    public function fields(Request $request)
+    {
+        return [
+            Text::make('First Name', 'name'),
+            Text::make('Last Name', 'name_last'),
+            DateTime::make('Signed', 'created_at'),
+            Boolean::make('Playing'),
+            BelongsTo::make('Participant', 'participant', nova('participant')),
+            BelongsTo::make('Room', 'room', nova('room')),
+            Date::make('Date of Birth', 'dob'),
+            Number::make('Minors'),
+            KeyValue::make('List of Minors', 'minors_names')->rules('json')
+            ->keyLabel('#')
+            ->valueLabel('Name')
+            ->resolveUsing(function ($minors) {
+                $collection = collect($minors);
+                $keyed = $collection->mapWithKeys(function ($item) {
+                    return [$item['id'] => $item['name']];
+                });
+
+                return $keyed->all();
+            }),
+            DateTime::make('Email Sent', 'emailed_at'),
+            Boolean::make('Valid Email'),
+            ID::make(),
+        ];
+    }
+
+    public function cards(Request $request)
+    {
+        return [];
+    }
+
+    public function filters(Request $request)
+    {
+        return [
+            new Filters\RoomLocation,
+            new Filters\Room,
+        ];
+    }
+
+    public function lenses(Request $request)
+    {
+        return [];
+    }
+
+    public function actions(Request $request)
+    {
+        return [];
+    }
```

Comment:
These can be refactored with the new abilities of the BaseResource. Check out the example here:

- https://github.com/tipoff/checkout/pull/22